roofit/roofit/src/RooPoisson.cxx

```diff
@@ -99,7 +99,7 @@ double RooPoisson::analyticalIntegral(Int_t code, const char* rangeName) const
     if (xmax < 0. || xmax < xmin) {
       return 0.;
     }
-    if (!x.hasMax() || RooNumber::isInfinite(xmax)) {
+    if ((!x.hasMax() || RooNumber::isInfinite(xmax)) && xmin <= 0.0) {
```

Comment:
Maybe we could use a cutoff (e.g. 100)  like : 
`xmin  < max( mu -  100*sqrt(mu),0) && xmax > mu + 100 * sqrt(mu)`

Reply:
Good idea! I will try this out